source/tutorial/gsfa/GSFA.rst

```diff
@@ -0,0 +1,414 @@
+-------------------
+
+Given a large number of high-dimensional labeled samples, supervised
+learning algorithms can often not be applied due to prohibitive
+computational requirements. In such cases, the following general scheme
+based on hierarchical GSFA is proposed:
+
+1\. Transform the labeled data to structured data, where the label information behind is implicitly encoded in the connections between the data points (samples).This permits using unsupervised learning algorithms, such as SFA, or its extension GSFA.
+
+2\. Use hierarchical processing to reduce the dimensionality, resulting in low-dimensional data with component similarities strongly dependent on the graph connectivity. Since the label information is encoded in the graph connectivity, the low-dimensional data is highly predictive of the labels.
+
+3\. Convert the (low-dimensional) data back to labeled data by combining the low-dimensional data points with the original labels or classes. This now constitutes a dataset suitable for standard supervised learning methods, because the dimensionality has become manageable.
+
+4\. Use standard supervised learning methods on the low-dimensional labeled data to estimate the labels. The unsupervised hierarchical network plus the supervised direct method together constitute the classifier or regression architecture.
+
+In the case of GSFA, the structured training data is called training
+graph, a weighted graph that has vertices representing the samples,
+vertex weights specifying a priori sample probabilities, and edge
+weights indicating desired output similarities, as derived from the
+labels.
+
+.. image:: approach.png
+    :width: 400px
+    :align: center
+    :height: 400px
+
+--------------
+
+.. _How the graphs are used for the training:
+
+3. How the graphs are used for the training 
+-------------------------------------------
+
+
+The training data is represented as a training graph
+:math:`G = (\textbf{V}, \textbf{E})` *(as illustrated in Figure bellow)*
+s.t. :
+
+-  :math:`\textbf{V}` corresponds to set of :math:`\textbf{x}(n)`
+   :math:`\rightarrow` *each vertex of the graph is a sample*
+
+-  Each edge of :math:`\textbf{E}` corresponds to a pair of samples
+   :math:`(\textbf{x}(n), \textbf{x}(n'))`
+
+
+Weights:
+^^^^^^^^
+
+1\. Edge weights indicate the **similarity between the connected vertices**. And since edges are undirected and have symmetric weights:
+
+   .. math:: \gamma_{n, n'} = \gamma_{n', n}
+
+2\. Each vertex  :math:`\textbf{x(}n\textbf{)}` has an associated weight :math:`v_n > 0`, which can be used to reflect its importance, frequency, or reliability.
+
+For instance, a sample occurring frequently in an observed phenomenon
+should have a larger weight than a rare sample. This representation
+includes the standard time series as a special case in which the graph
+has a linear structure and all node and edge weights are identical *(as
+illustrated in Figure(b))*.
+
+.. image:: training_graph.png
```

Comment:
Some images appear not to be centered correctly and hard to read. Maybe there is a way to change this. If you need some help in plotting, let me know.